tests/data/test_history.py

```diff
@@ -129,16 +130,17 @@ def test_load_data_with_new_pair_1min(ticker_history_list, mocker, caplog,
     )
 
     # download a new pair if refresh_pairs is set
-    load_pair_history(datadir=testdatadir, timeframe='1m',
-                      refresh_pairs=True, exchange=exchange, pair='MEME/BTC')
+    refresh_data(datadir=testdatadir, timeframe='1m', pairs=['MEME/BTC'],
+                 exchange=exchange)
+    load_pair_history(datadir=testdatadir, timeframe='1m', pair='MEME/BTC')
     assert file.is_file()
     assert log_has_re(
         'Download history data for pair: "MEME/BTC", timeframe: 1m '
         'and store in .*', caplog
     )
     with pytest.raises(OperationalException, match=r'Exchange needs to be initialized when.*'):
-        load_pair_history(datadir=testdatadir, timeframe='1m',
-                          refresh_pairs=True, exchange=None, pair='MEME/BTC')
+        refresh_data(datadir=testdatadir, timeframe='1m', pairs=['MEME/BTC'],
+                     exchange=None)
```

Comment:
Not "really" in this line - but related...

Since `_download_pair_data()` is now only called from 2 places, both which have exchange as mandatory object - i think it's safe to remove this section of the test, as well as the `Optional[Exchange]` and the test for the exchange - since this can realistically never be called.

Unless I import the function manually and pass it something unfitting (a string instead of an exchange object - but that isn't checked anyway).
However,  that'll break most other functions too (we almost always rely on the type passed in to be correct ... otherwise we'd have every function first verify every object's type...) - and so do most other libraries.

Reply:
Actually I made it during last minor set of changes into data.history, removed Optional, but it led into mypy/flake complaints about the types (this `if not exchange` made mypy happy). I found then that it's easier and more effective to split this functionality first. So this PR was issued. after first part of the changes (#2672). Third part will follow...